immut/hashmap/hashmap.mbti

```diff
@@ -27,21 +43,34 @@ fn op_get[K : Eq + Hash, V](T[K, V], K) -> V?
 
 fn remove[K : Eq + Hash, V](T[K, V], K) -> T[K, V]
 
+fn singleton[K, V](K, V) -> T[K, V]
+
 fn size[K, V](T[K, V]) -> Int
 
+fn to_array[K, V](T[K, V]) -> Array[(K, V)]
+
 fn union[K : Eq + Hash, V](T[K, V], T[K, V]) -> T[K, V]
 
 // Types and methods
 type T[K, V]
 impl T {
   add[K : Eq + Hash, V](Self[K, V], K, V) -> Self[K, V]
+  contains[K : Eq + Hash, V](Self[K, V], K) -> Bool
   each[K, V](Self[K, V], (K, V) -> Unit) -> Unit
+  elems[K, V](Self[K, V]) -> Array[V]
+  filter[K : Eq + Hash, V](Self[K, V], (V) -> Bool) -> Self[K, V]
   find[K : Eq + Hash, V](Self[K, V], K) -> V?
+  fold[K, V, A](Self[K, V], init~ : A, (A, V) -> A) -> A
+  fold_with_key[K, V, A](Self[K, V], (A, K, V) -> A, init~ : A) -> A
   iter[K, V](Self[K, V]) -> Iter[(K, V)]
   iter2[K, V](Self[K, V]) -> Iter2[K, V]
+  keys[K, V](Self[K, V]) -> Array[K]
```

Comment:
Returns `Iter[K]` instead of `Array[K]`. About the signature, the one `Map::keys` is preferred, and `@sorted_map.T::keys` and `@immut/sorted_map.T::keys` will migrate to `Iter[K]` in the future.